### DISCUSSION TITLE
Is there a problem with the LFP cell parameter set（Prada2013）?

### DISCUSSION THREAD

**Discussion by RobinBluehub** [pybamm-team/PyBaMM#2661] [Q&A]
When I run the code like below, the parameter set works.
`import pybamm
parameter_values = pybamm.ParameterValues("Prada2013")
model = pybamm.lithium_ion.DFN()
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve([0, 3600])
sim.plot( )`
But when I join the experiment, the Prada2013 parameter set cannot run successfully, the code is as follows:
`import pybamm
pybamm.set_logging_level("NOTICE")
options = {
    #"thermal": "lumped",
    }
model = pybamm.lithium_ion.DFN(options=options)
parameter_values = pybamm.ParameterValues("Prada2013")
parameter_values.set_initial_stoichiometries(1)

solver = pybamm.CasadiSolver()
var_pts = {"x_n": 40, "x_s": 30, "x_p": 40, "r_n": 40, "r_p": 40}
experiment=pybamm.Experiment([(
        "Discharge at 1 C until 2.5 V",
        "Charge at 1 C until 3.65 V",
        "Hold at 3.65 V until C/10",
        "Hold at 3.0V until C/50",
    )]*1)

sim_1 = pybamm.Simulation(model,parameter_values=parameter_values,experiment=experiment,solver=solver,var_pts=var_pts)
sol_1=sim_1.solve(calc_esoh=False,initial_soc=1)
sim_1.plot(["X-averaged cell temperature [K]","Terminal voltage [V]","Current [A]","C-rate"])
`
There are warning and error.
1.warning:UserWarning: Q_Li=2.4127 Ah is greater than Q_p=1.6292 Ah.
  warnings.warn(f"Q_Li={Q_Li:.4f} Ah is greater than Q_p={Q_p:.4f} Ah.")
2.error:SolverError: Could not find acceptable solution: solver terminated successfully, but maximum solution error (1.6999898910796674e-06) above tolerance (1e-06)
This problem has troubled me for a long time, and I hope the pybamm team can take the time to answer my doubts.
Best regards
          Robin

**Comment by brosaplanella** [comment]
Hi! The Prada parameter set has been taken from the article straight away and we have not personally validated so there might be some issues. I would suggest going back to the original source to check the details of the parameter set.

**Reply by RobinBluehub** [reply]
I found that the Prada parameter set is composed of different types of battery parameters in multiple articles, which may be the reason why the experiment I designed cannot be carried out.

**Reply by 1836005678** [reply]
I have also used the Prada2013 parameter set recently as it is the only LiFePO4 positive electrode battery parameter set in Pybamm, and i also find the '2.error:SolverError: Could not find acceptable solution: solver terminated successfully, but maximum solution error (1.6999898910796674e-06) above tolerance (1e-06)'.
Actually, it comes from the sentence 'initial_soc=1'. If we change the default tolerance (1e-06) to (10) in 'pybamm\solvers\algebraic_solver.py' in 'class AlgebraicSolver(pybamm.BaseSolver):' in 'def __init__(self, method="lm", tol=1e-06, extra_options=None):', then this bug will be fixed. However, the initial output Voltage will be much lower(about 3.2V) than normal settings(about 3.6V).
I have been tring to fix this bug without changing the tolerance. However, as the code calls too many functions, I can't find where the problem is.
Another way to get around the bug is to calculate the initial stoichiometric number manually. However, I do not know what the formula can we use.
I would be very happy if you would like to discuss with me or help me solve the problem.

**Reply by RobinBluehub** [reply]
Sorry, I'm a beginner at pybamm. I'm afraid I can't help you.

**Reply by 1836005678** [reply]
I come to realize that the '2.error' had been fixed by #1811 , especially [.](https://github.com/pybamm-team/PyBaMM/commit/eabb72040892b964907e01cdc09130a7e25a1489) To fixed the problem, just change 7 parameters (see 7 green rows from the link) and do not use `calc_esoh=False` item when call the function '.solve', just like `sol_1=sim_1.solve(calc_esoh=False,initial_soc=1)`. Simulation show that the output Voltage is 4.25391528648289V (Upper Voltage cut-off is 4.4V) when discharge begins and 1C discharge takes 3478.86661547256s (0.966h). This result is proper. (As the OCP curve is extremely steep when SOC is close to 100%, 4.2539V is understandable).

**Reply by ddagrinho** [reply]
can you please explain how you solve the problem? because when i am following the link i can not find the parameters that you change

**Reply by ALMA7867** [reply]
Try with changing the solver (use [IDAKLU Solver])

**Comment by ganmad** [comment]
It seems Prada2013 LFP parameters are not consistent. Here is a thesis: https://kth.diva-portal.org/smash/get/diva2:1593557/FULLTEXT02.pdf
that covers an updated set of parameters for same LFP cell, measured from the cell teardown. The author used PyBaMM  to simulate the cell. Hope this will be useful.

**Reply by brosaplanella** [reply]
That's great (sorry I just saw this)! Was this implemented in #3096 @tinosulzer? If not, that would be a great first issue if you want to give it a go @ganmad

**Reply by ganmad** [reply]
Thanks for linking to the issue. I will give it a try sometime.

**Reply by Thiyaga2025** [reply]
@brosaplanella and @ganmad ,

is the current 'Prada2013' parameter set updated one or should i pickup parameters from other sources mentioned here.

thanks.

**Reply by brosaplanella** [reply]
I believe it is the updated one, but if you can check and let us know if that is not the case it would be great!

**Reply by Thiyaga2025** [reply]
@brosaplanella ,

thanks for the reply. I am running some simulations for LFP chemistry, will update issues, if any.

**Reply by JuanCarlosZapata666** [reply]
@Thiyaga2025  did you find any issues?
